core_v2/sources/domains.move

```diff
@@ -156,28 +146,121 @@ module aptos_names::domains {
     }
 
-    public entry fun init_reverse_lookup_registry_v1(account: &signer) {
-        assert!(signer::address_of(account) == @aptos_names, error::permission_denied(ENOT_AUTHORIZED));
+    public fun token_addr(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): address acquires CollectionCapabilityV2 {
+        token::create_token_address(
+            &get_token_signer_address(),
+            &config::collection_name_v1(),
+            &token_helper::get_fully_qualified_domain_name(subdomain_name, domain_name),
+        )
+    }
 
-        if (!exists<ReverseLookupRegistryV1>(@aptos_names)) {
-            move_to(account, ReverseLookupRegistryV1 {
-                registry: table::new()
-            });
+    public fun get_record_obj(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): Object<NameRecordV2> acquires CollectionCapabilityV2 {
+        object::address_to_object(token_addr_inline(domain_name, subdomain_name))
+    }
 
-            move_to(account, SetReverseLookupEventsV1 {
-                set_reverse_lookup_events: account::new_event_handle<SetReverseLookupEventV1>(account),
-            });
+    inline fun get_record(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): &NameRecordV2 acquires CollectionCapabilityV2 {
+        borrow_global(token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun get_record_mut(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): &mut NameRecordV2 acquires CollectionCapabilityV2 {
+        borrow_global_mut(token_addr_inline(domain_name, subdomain_name))
+    }
+
+    /// Creates a token for the name.
+    /// NOTE: This function performs no validation checks
+    fun create_token(
+        to_addr: address,
+        domain_name: String,
+        subdomain_name: Option<String>,
+        expiration_time_sec: u64,
+    ) acquires CollectionCapabilityV2 {
+        let name = token_helper::get_fully_qualified_domain_name(subdomain_name, domain_name);
+        let description = config::tokendata_description();
+        let uri: string::String = config::tokendata_url_prefix();
+        string::append(&mut uri, name);
+        let constructor_ref = token::create_named_token(
+            &get_token_signer(),
+            config::collection_name_v1(),
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+        let token_signer = object::generate_signer(&constructor_ref);
+        let record = NameRecordV2 {
+            domain_name,
+            subdomain_name,
+            expiration_time_sec,
+            target_address: option::none(),
+            transfer_ref: object::generate_transfer_ref(&constructor_ref),
         };
+        move_to(&token_signer, record);
+        let record_obj = object::object_from_constructor_ref<NameRecordV2>(&constructor_ref);
+        object::transfer(&get_token_signer(), record_obj, to_addr);
```

Comment:
is this how it works? 
1. line 254 moves the created record to function caller
2. line 255 get `Object<NameRecordV2>`
3. line 256 transfer object to `to_addr`

why can't we move the record to `to_addr` directly without passing through signer

Reply:
also why can't we get the `record_obj` like [hero.move](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/token_objects/hero/sources/hero.move#L120)
```let record_obj = object::address_to_object(signer::address_of(&token_signer));```